core/src/main/java/com/oath/oak/NativeAllocator/OakNativeMemoryAllocator.java

```diff
@@ -57,83 +71,83 @@
     // constructor
     // input param: memory capacity given to this Oak. Uses default BlocksPool
     public OakNativeMemoryAllocator(long capacity) {
-      this(capacity, BlocksPool.getInstance());
+        this(capacity, BlocksPool.getInstance());
     }
 
     // A testable constructor
     OakNativeMemoryAllocator(long capacity, BlocksProvider blocksProvider) {
         this.blocksProvider = blocksProvider;
-        int blockArraySize = ((int)(capacity/blocksProvider.blockSize()))+1;
+        int blockArraySize = ((int) (capacity / blocksProvider.blockSize())) + 1;
         // first entry of blocksArray is always empty
-        this.blocksArray = new Block[blockArraySize+1];
+        this.blocksArray = new Block[blockArraySize + 1];
         // initially allocate one single block from pool
         // this may lazy initialize the pool and take time if this is the first call for the pool
         allocateNewCurrentBlock();
         this.capacity = capacity;
+        for (int i = 0; i < ThreadIndexCalculator.MAX_THREADS; i++) {
+            dummies[i] = new FreeChuck(-1, 0, null);
+        }
     }
 
     @Override
     public ByteBuffer allocate(int size) {
-      return allocate(size, true);
+        return allocateSlice(size).getByteBuffer();
```

Comment:
I allocate() method in use anywhere? Is it @Override (I think we changed the interface)?

Reply:
It is now deprecated. It is used in some tests so I left it in the code.